### PR TITLE
Toml

### LINKED ISSUE #33: Add a config.yaml for input arguments
Right now, we have CLI arguments and we have environment arguments. I think that it would be best to implement inputs in a uniform way that is also a good experience.

I think the best way around this is to use CLI arguments with defaults but allow the user to load a `config.yaml` file. This would be the same set up as [Erigon](https://github.com/ledgerwatch/erigon). 

The main tasks are then:
- [x] Allow for a flag like `--config-file` where a user can provide a `config.yaml`. 
- [x] Set up a preconfigured `config.yaml` file.
- [x] Change API/RPC away from environment variables and into CLI configs.

---

What do you think? @0xJepsen  @eljhfx @ts0yu

### PR DESCRIPTION
**Pull request by 0xJepsen**
This PR adds support for a configuration file and closes #33 
the PR includes additions to the README on how to use the configuration file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,22 +105,35 @@ Liquidity: 21972098821216706277
 Tick:      205282
 Price:     "1.216568804789000000000000000000000000000e+3"
 ```
+
 You may also build the executable with `cargo build`, which will output a binary in `target/`
 
+## Configuration File
+
+To run arbiter with a configuration file, we can run the command `cargo run see --config true` or, if we have built the latest changes, `arbiter see --config true`. This will load in configuration settings from a `config.toml` file located in `./crates/cli/src/`. Feel free to make custom configuration files for your needs. The logic that finds the file is in `crates/cli/src/config/mod.rs`; remember to add the path of your new file to the list of file paths.
```
Comment (ts0yu) [inline README.md:113]: We need to update this so it reflects the latest changes of being able to run `arbiter see --config`, without the true parameter.

File: crates/clairvoyance/src/uniswap.rs [hunk 2]
```diff
@@ -88,6 +92,26 @@ impl Pool {
...
 }
 
+// Creates a pool from the cli/or config parameters
+pub async fn get_pool(
+    token0: &String,
+    token1: &String,
+    bp: &str,
+    provider: Arc<Provider<Http>>,
+) -> Result<Pool, Box<dyn Error>> {
+    // let provider = match env::var_os("PROVIDER") {
```
Comment (ts0yu) [inline crates/clairvoyance/src/uniswap.rs:102]: let's remove the comments before we merge with main

File: crates/cli/src/config/mod.rs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                see: None,
+            }
+        });
+        // ! check the rpc_url field in the network config.toml section
+        let rpc_url = match config_toml.network {
+            Some(network) => network.rpc_url.unwrap_or_else(|| {
+                println!("Missing Field rpc_url in network config");
+                "unknown".to_owned()
+            }),
+            None => "unknown".to_owned(),
```
Comment (ts0yu) [inline crates/cli/src/config/mod.rs:63]: Setting the field to `"unknown"` is basically just letting the program panic at a later stage so its basically redundant here imo

File: crates/cli/src/config/mod.rs [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+        };
+        // ! check the token0, token1, and bp fields in the see config.toml section
+        let (token0, token1, bp) = match config_toml.see {
+            Some(see) => {
+                let token0 = see.token0.unwrap_or_else(|| {
+                    println!("Missing Field token0 in see config");
+                    "unknown".to_owned()
```
Comment (ts0yu) [inline crates/cli/src/config/mod.rs:70]: same here and below

File: crates/cli/src/main.rs [hunk 5]
```diff
@@ -62,31 +60,37 @@ async fn main() -> Result<()> {
...
-            for pool in pools {
-                join!(pool.monitor_pool());
+            match config {
+                true => {
+                    // If present, load config.toml and get pool from there
+                    println!("Loading config.toml...");
+                    let config_obj = config::Config::new();
+                    println!("Config.toml loaded successfully! Getting Pool...");
+                    let pool = get_pool(
+                        &config_obj.token0,
```
Comment (ts0yu) [inline crates/cli/src/main.rs:70]: the load will appear to be successful when in fact `token0` is `"unknown"`

**Review by ts0yu — CHANGES_REQUESTED** [review]
looking good! just a few concerns about the error handling, but otherwise solid stuff.
i can make some changes for custom error handling with `thiserror` tomorrow, if you want

File: crates/cli/src/config/mod.rs (hunk 3 above)
Comment (0xJepsen) [reply crates/cli/src/config/mod.rs:63]: This would be pretty good imo

File: crates/cli/src/main.rs (hunk 5 above)
Comment (0xJepsen) [reply crates/cli/src/main.rs:70]: Good point, i replaced with a more correct statement

**Review by ts0yu — APPROVED** [review]

**Pull request merged**